lib/config_reader.dart

```
@@ -0,0 +1,42 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:flutter/services.dart';
+
+/// - contains the initalization for the firebase emulator suite
+///- Added only for mobile platforms
+///- will only return a singleton instance
+/// - this will get all the data required to setup from
+/// `./configs/emulator_config.json` if this is not added will use default settings
+class ConfigReader {
+  factory ConfigReader() => _instance;
+  ConfigReader._();
+  static final _instance = ConfigReader._();
+  final Map<String, dynamic> _config = {};
+  Future<void> initializeConfigReader() async {
+    try {
+      final configurationOpitons = await rootBundle.loadString('configs/emulator_config.json');
+      final jsonValue = jsonDecode(configurationOpitons) as Map<String, dynamic>;
+      _config.addAll(jsonValue);
+    } on Exception catch (_) {
+      print('initializeConfigReader | Error | No emulator_config.json found using defaults');
```

What happens if this case occurs? Do we just not initialize the emulator and if so is this expected in the case the developer wants to use their own firebase backend?

Or should we just let this throw an exception so that people don't miss this message?

---

I think we dont need , some if they dont change the hosts and ports this step is not needed.
Else we could also throw to stop the debugger , both are applicable

---

So in which scenarios does this print actually happen? What would be the drawback of just throwing an exception?

---

Very well then lets throw the exception, also one to note it can be ignored too